tests/helpers.py

```diff
@@ -0,0 +1,129 @@
+import sqlite3
+from abc import ABC, abstractmethod
+
+import pandas as pd
+
+import splink.duckdb.duckdb_comparison_level_library as cll_duckdb
+import splink.duckdb.duckdb_comparison_library as cl_duckdb
+import splink.spark.spark_comparison_level_library as cll_spark
+import splink.spark.spark_comparison_library as cl_spark
+import splink.sqlite.sqlite_comparison_level_library as cll_sqlite
+import splink.sqlite.sqlite_comparison_library as cl_sqlite
+from splink.duckdb.duckdb_linker import DuckDBLinker
+from splink.spark.spark_linker import SparkLinker
+from splink.sqlite.sqlite_linker import SQLiteLinker
+
+
+class TestHelper(ABC):
+    @property
+    @abstractmethod
+    def Linker(self):
+        pass
+
+    def extra_linker_args(self):
+        return {}
+
+    @abstractmethod
+    def convert_frame(self, df):
+        pass
+
+    def load_frame_from_csv(self, path):
+        return pd.read_csv(path)
+
+    @property
+    @abstractmethod
+    def cll(self):
+        pass
+
+    @property
+    @abstractmethod
+    def cl(self):
+        pass
+
+
+class DuckDBTestHelper(TestHelper):
+    @property
+    def Linker(self):
+        return DuckDBLinker
+
+    def convert_frame(self, df):
+        return df
+
+    @property
+    def cll(self):
+        return cll_duckdb
+
+    @property
+    def cl(self):
+        return cl_duckdb
+
+
+class SparkTestHelper(TestHelper):
+    def __init__(self, spark):
+        self.spark = spark
+
+    @property
+    def Linker(self):
+        return SparkLinker
+
+    def extra_linker_args(self):
+        return {"spark": self.spark}
+
+    def convert_frame(self, df):
+        spark_frame = self.spark.createDataFrame(df)
+        spark_frame.persist()
+        return spark_frame
+
+    def load_frame_from_csv(self, path):
+        df = self.spark.read.csv(path, header=True)
+        df.persist()
+        return df
```

Comment:
We might as well add a `load_frame_from_parquet` method here too.

Reply:
added